deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/impl/KubernetesDeployer.java

```diff
@@ -164,11 +180,21 @@ public void unDeploy(DeploymentContext context) throws HyscaleException {
         }
     }
 
-    @Override
-    public boolean authenticate(AuthConfig authConfig) {
-        // TODO
-        return false;
-    }
+	@Override
+	public boolean authenticate(AuthConfig authConfig) throws HyscaleException {
+		ApiClient apiClient = null;
+		apiClient = clientProvider.get((K8sAuthorisation) authConfig);
+		AuthenticationV1Api apiInstance = new AuthenticationV1Api(apiClient);
+		try {
+			V1APIResourceList result = apiInstance.getAPIResources();
+			System.out.println(result);
+			return true;
+		} catch (ApiException e) {
+			logger.error("Exception when calling AuthenticationV1Api#createTokenReview");
+		}
+		return false;
+	}
```

Comment:
This should be moved to AuthenticationHandler on lines on NamespaceHandler which should return boolean in case of valid and invalid. And throw HyscaleException in case we get ApiException.